src/typeMapper.js

```diff
@@ -0,0 +1,58 @@
+import { _ } from 'gridjs-react';
+
+export const typeRepresentationMapper = {
+  "img": (value) => _(<img src={value}></img>),
+  "int": (value) => {return literalToNumber(value, parseInt) },
+  "float": (value) => {return literalToNumber(value, parseFloat)},
+  "euro": (value) => {return `€ ${getLiteralValue(value)}`},
+  "dollar": (value) => {return `$ ${getLiteralValue(value)}`}
+}
+
+export const typeSortMapper = {
+  "int": sortNumber,
+  "float": sortNumber
+}
+
+/**
+ * Comperator function for any subclass of Number. 
+ * @param {Number} valueA  
+ * @param {Number} valueB 
+ */
+function sortNumber(valueA, valueB){
+  if(valueA > valueB || isNaN(valueB)){
+    return 1
+  }
+  else if(valueB > valueA || isNaN(valueA)){
+    return -1
+  }
+  else{
+    return 0
+  }
+}
+
+/**
+ * 
+ * @param {Number} value string value which is to be converted to a number
+ * @param {Function} parseFunction a function that takes the string value of a number and converts it to the corresponding number subclass 
+ * @returns A number parsed in the correct type (e.g. integer, float...) or an empty string (instead of NaN)
+ */
+function literalToNumber(value, parseFunction){
+  let literalValue = getLiteralValue(value)
+  if(literalValue){
```

Comment:
What happens if the literal value is `""`?

Reply:
But the function claims to return a number and string is not a number. So either the function doc needs to change or the function should throw an error.

Reply:
I have changed this to return undefined instead of an empty string in [commit 85429a3](https://github.com/SolidLabResearch/generic-data-viewer/pull/6/commits/85429a36a3fe7d215d11bfad0ac3094c68afed17)